src/tslintMulti.ts

```diff
@@ -89,54 +90,34 @@ class MultiLinter {
+        let enabledRules = this.getEnabledRules(fileName, source, configuration);
+
+        if (this.options.fix) {
+            this.fixes = [];
+            for (let rule of enabledRules) {
+                let fileFailures = this.applyRule(rule, sourceFile);
+                const fixes = fileFailures.map(f => f.getFix()).filter(f => !!f);
+                let content = fs.readFileSync(fileName, { encoding: "utf-8" });
+                if (fixes.length > 0) {
+                    this.fixes = this.fixes.concat(fileFailures);
+                    content = Fix.applyAll(content, fixes);
+                    fs.writeFileSync(fileName, content, { encoding: "utf-8" });
+
+                    // reload AST if file is modified
+                    sourceFile = this.getSourceFile(fileName, content);
+                    enabledRules = this.getEnabledRules(fileName, content, configuration);
+                }
+                this.failures = this.failures.concat(fileFailures);
             }
-        } else {
-            sourceFile = getSourceFile(fileName, source);
         }
 
-        if (sourceFile === undefined) {
-            throw new Error(`Invalid source file: ${fileName}. Ensure that the files supplied to lint have a .ts, .tsx, or .js extension.`);
-        }
-
-        // walk the code first to find all the intervals where rules are disabled
-        const rulesWalker = new EnableDisableRulesWalker(sourceFile, {
-            disabledIntervals: [],
-            ruleName: "",
-        });
-        rulesWalker.walk(sourceFile);
-        const enableDisableRuleMap = rulesWalker.enableDisableRuleMap;
-
-        const rulesDirectories = arrayify(this.options.rulesDirectory)
-            .concat(arrayify(configuration.rulesDirectory));
-        const configurationRules = configuration.rules;
-        const jsConfiguration = configuration.jsRules;
-        const isJs = fileName.substr(-3) === ".js";
-        let configuredRules: IRule[];
-
-        if (isJs) {
-            configuredRules = loadRules(jsConfiguration, enableDisableRuleMap, rulesDirectories, true);
-        } else {
-            configuredRules = loadRules(configurationRules, enableDisableRuleMap, rulesDirectories, false);
-        }
-
-        const enabledRules = configuredRules.filter((r) => r.isEnabled());
-        for (let rule of enabledRules) {
-            let ruleFailures: RuleFailure[] = [];
-            if (this.program && rule instanceof TypedRule) {
-                ruleFailures = rule.applyWithProgram(sourceFile, this.program);
-            } else {
-                ruleFailures = rule.apply(sourceFile);
-            }
-            for (let ruleFailure of ruleFailures) {
-                if (!this.containsRule(this.failures, ruleFailure)) {
-                    this.failures.push(ruleFailure);
-                }
+        // make a 2nd pass if there were any fixes because the positions may be off        
+        if (!this.options.fix || this.fixes.length > 0) {
```

Comment:
the first part of this conditional expression looks unintuitive to me -- don't you want to only continue if `this.options.fix` is true?